### DISCUSSION TITLE
Passing data between GD.yield() and resume()

### DISCUSSION THREAD

**Discussion by Alikae** [gilzoide/godot-lua-pluginscript#13] [Q&A] [answered]
Hi !
I would like to know if there is currently a way to pass data between between GD.yield() and the corresponding resume().
All my tests so far behave as if GD.yield() always return nil, regardless of how I call resume().

Wanted usage:
script a:
> b.test()
> b.resume('data')

 script b:
> function test()
> a = GD.yield()
> print(a) -- print data

Is there something I missed about how to use it ?
Is this maybe not an intended feature ? The wanted behavior is easily reproduced by changing
> b.resume('data')

by
> b.some_prop = 'data'
> b.resume()

Lastly, I came to the sources to try to understand it by myself, but I stumbled upon coroutine.resume and coroutine.yield, that (in my understanding) doesn't seems to be defined anywhere. Does it come from Godot API ? If so, do you know where I can find some documentation about it ? Pretty hard to find Godot documentation that do more than scratching the surface.

**Comment by gilzoide** [comment, marked as answer]
Hi there!
I thought data was already being passed from `resume` to the code that `yield`ed, but I'll be honest and say that I haven't tested this flow.
`coroutine.resume` and `coroutine.yield` are builtin Lua functions, part of its [coroutine implementation](https://www.lua.org/manual/5.1/manual.html#2.11).

> Is there something I missed about how to use it ?

Not at all, this is a bug.
Most likely the fault is the lack of `return`ing the result of `coroutine.yield` at the end of `GD.yield`: https://github.com/gilzoide/godot-lua-pluginscript/blob/b2100ffaf54ec24d4ba392d5fac57c83323f09e5/src/late_globals.lua#L132-L133
Sorry about that.

I created issue #14 related to this problem. Thanks for the report!

**Comment by gilzoide** [comment]
#15 should fix this.
If you want, you can download the zip generated by GitHub Actions (https://github.com/gilzoide/godot-lua-pluginscript/actions/runs/1761324506) and install this fixed version in your project. If you do try it out, please tell me if it works ^^
